internal/provider/resource_azure_peering_connection.go

```diff
@@ -0,0 +1,400 @@
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func resourceAzurePeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description: "The Azure peering connection resource allows you to manage a peering connection between an HVN and a peer Azure VNet.",
+
+		CreateContext: resourceAzurePeeringConnectionCreate,
+		ReadContext:   resourceAzurePeeringConnectionRead,
+		DeleteContext: resourceAzurePeeringConnectionDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &peeringDefaultTimeout,
+			Create:  &peeringCreateTimeout,
+			Delete:  &peeringDeleteTimeout,
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAzurePeeringConnectionImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn_link": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"peering_id": {
+				Description:      "The ID of the peering connection.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"peer_vnet_name": {
+				Description: "The name of the peer VNet in Azure.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"peer_subscription_id": {
+				Description: "The subscription ID of the peer VNet in Azure.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"peer_vnet_region": {
+				Description: "The region of the peer VNet in Azure.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
+					return strings.EqualFold(old, new)
+				},
+			},
+			"peer_tenant_id": {
+				Description: "The tenant ID of the peer VNet in Azure.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"peer_resource_group_name": {
+				Description: "The resource group name of the peer VNet in Azure.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			// Computed outputs
+			"organization_id": {
+				Description: "The ID of the HCP organization where the peering connection is located. Always matches the HVN's organization.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the HCP project where the peering connection is located. Always matches the HVN's project.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"application_id": {
+				Description: "The application ID of the HCP VNet backing the HVN.",
```

Comment:
Likewise here too: "The ID of Azure application whose credentials are used to peer VNets."